controllers/cbcontainersagent_controller_test.go

```diff
@@ -77,19 +87,26 @@ func testCBContainersClusterController(t *testing.T, setups ...SetupClusterContr
 		Scheme:    &runtime.Scheme{},
 		Namespace: agentNamespace,
 
-		ClusterProcessor: mocksObjects.ClusterProcessor,
-		StateApplier:     mocksObjects.StateApplier,
+		ClusterProcessor: mocksObjects.clusterProcessor,
+		StateApplier:     mocksObjects.stateApplier,
 	}
 
 	return controller.Reconcile(mocksObjects.ctx, ctrlRuntime.Request{})
 }
 
-func setupClusterCustomResource(testMocks *ClusterControllerTestMocks) {
-	testMocks.client.EXPECT().List(testMocks.ctx, &cbcontainersv1.CBContainersAgentList{}).
-		Do(func(ctx context.Context, list *cbcontainersv1.CBContainersAgentList, _ ...interface{}) {
-			list.Items = ClusterCustomResourceItems
-		}).
-		Return(nil)
+func setupClusterCustomResource(items ...cbcontainersv1.CBContainersAgent) SetupClusterControllerTest {
+	if len(items) == 0 {
+		items = make([]cbcontainersv1.CBContainersAgent, len(ClusterCustomResourceItems))
```

Comment:
I did this because it turned out all our tests modify the shared global var due to pointers. This ensures the reconciler gets a **copy** of the data so changes to the CRs in one test are not reflect in the others.